libbeat/setup/kibana/client.go

```diff
@@ -0,0 +1,141 @@
+package kibana
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type Connection struct {
+	URL     string
+	Headers map[string]string
+
+	http    *http.Client
+	version string
+}
+
+type Client struct {
+	Connection
+}
+
+func addPath(_url, _path string) (string, error) {
+
+	u, err := url.Parse(_url)
+	if err != nil {
+		return "", err
+	}
+	u.Path = path.Join(u.Path, _path)
+	return u.String(), nil
+}
+
+func NewKibanaClient(cfg *common.Config) (*Client, error) {
+
+	config := defaultKibanaConfig
+	if err := cfg.Unpack(&config); err != nil {
+		return nil, err
+	}
+
+	kibanaURL, err := common.MakeURL(config.Protocol, config.Path, config.Host, 5601)
+	if err != nil {
+		logp.Err("Invalid host param set: %s, Error: %v", config.Host, err)
+		return nil, fmt.Errorf("invalid Kibana host: %v", err)
+	}
+
+	logp.Debug("kibana", "Kibana url: %s", kibanaURL)
+
+	client := &Client{
+
+		Connection: Connection{
+			URL: kibanaURL,
+			http: &http.Client{
+				Transport: &http.Transport{
+					TLSClientConfig: &tls.Config{InsecureSkipVerify: true}, // ignore expired SSL certificates
+				},
+			},
+		},
+	}
+	return client, nil
+}
+
+func (conn *Connection) Request(method, extraPath string, params url.Values, body io.Reader) (int, []byte, error) {
+
+	reqURL, err := addPath(conn.URL, extraPath)
+	if err != nil {
+		return 0, nil, fmt.Errorf("fail to parse URL %s: %v", conn.URL, err)
```

Comment:
Would be perhaps better to write the "fail to parse URL" error in the `addPath` method?